scipy/stats/_stats_py.py

```diff
@@ -11008,3 +11008,126 @@ def first_order(t):
     # finding a wrong root.
     res = root_scalar(first_order, x0=x0, x1=x1)
     return res.root
+
+def lmoments_iv(sample, n_moments, axis, sorted, standardize):
+    # If n_moments is invalid, passing `dtype=int` would make `asarray` fail.
+    # We want to raise a more readable error message.
+
+    n_moments = np.asarray(n_moments)[()]
+    message = "`n_moments` must be a positive integer."
+    if (not np.issubdtype(n_moments.dtype, np.integer) or n_moments.ndim != 0
+            or n_moments <= 0):
+        raise ValueError(message)
+
+    axis = np.asarray(axis)[()]
+    message = "`axis` must be an integer."
+    if not np.issubdtype(axis.dtype, np.integer) or axis.ndim != 0:
+        raise ValueError(message)
+
+    sorted = np.asarray(sorted)[()]
+    message = "`sorted` must be True or False."
+    if not np.issubdtype(sorted.dtype, np.bool_) or sorted.ndim != 0:
+        raise ValueError(message)
+
+    standardize = np.asarray(standardize)[()]
+    message = "`standardize` must be True or False."
+    if not np.issubdtype(standardize.dtype, np.bool_) or standardize.ndim != 0:
+        raise ValueError(message)
+
+    sample = np.moveaxis(sample, axis, -1)
+    sample = np.sort(sample, axis=-1) if not sorted else sample
+
+    return sample, n_moments, axis, sorted, standardize
+
+
+def _br(x, *, r=0):
+    n = x.shape[-1]
+    x = np.expand_dims(x, axis=-2)
+    x = np.broadcast_to(x, x.shape[:-2] + (len(r), n))
+    x = np.triu(x)
+    j = np.arange(n)
+    return np.sum(special.binom(j, r[:, np.newaxis])*x, axis=-1) / special.binom(n-1, r) / n
```

Comment:
In the paper, this appears as 
<img width="1006" alt="image" src="https://github.com/scipy/scipy/assets/6570539/0694f1a3-d8a9-412e-a27d-81e1543b8811">

It is a little tricky to compare this against the formula. Without thinking carefully:
- It may look as though `special.binom(j, ...)` should be something like `special.binom(j - 1, ...)`
- It may look as though `j = np.arange(n)` should be something like `j = np.arange(r, n + 1)`

But I think closer will inspection will reveal that this is equivalent. Note that `r` is a vector, and the formula is implemented this way to avoid explicit looping over the elements. The `x = np.triu(x)` zeros out the terms that shouldn't be part of the sum.